source/grid/grid_generator.cc

```diff
@@ -1910,6 +1910,353 @@ namespace GridGenerator
     tria.create_triangulation(points, cells, SubCellData());
   }
 
+
+
+  template <>
+  void plate_with_a_hole(Triangulation<1> & /*tria*/,
+                         const double /*inner_radius*/,
+                         const double /*outer_radius*/,
+                         const double /*pad_bottom*/,
+                         const double /*pad_top*/,
+                         const double /*pad_left*/,
+                         const double /*pad_right*/,
+                         const Point<1> /*center*/,
+                         const types::manifold_id /*polar_manifold_id*/,
+                         const types::manifold_id /*tfi_manifold_id*/,
+                         const double /*L*/,
+                         const unsigned int /*n_slices*/,
+                         const bool /*colorize*/)
+  {
+    Assert(false, ExcNotImplemented());
+  }
+
+
+  namespace internal
+  {
+    // helper function to check if point is in 2D box
+    bool inline point_in_box(const Point<2> &p,
+                             const Point<2> &c,
+                             const double    radius)
+    {
+      return (std::abs(p[0] - c[0]) < radius) &&
+             (std::abs(p[1] - c[1]) < radius);
+    }
+
+    bool inline point_in_box(const Point<3> &p,
+                             const Point<3> &center,
+                             const double    radius)
+    {
+      return point_in_box(Point<2>(p[0], p[1]),
```

Comment:
Hmm... I don't find this to be very intuitive. It makes sense in 2d, but in 3d you're taking the projection down into 2d so you're not considering a box but rather a z-aligned infinitely long square tube. I think that you could probably replace all instances of this function with the [`BoundingBox`](https://dealii.org/developer/doxygen/deal.II/classBoundingBox.html#a8a33b5a017e2f70213bce0bcf447c13e) class.

Reply:
i don't want it to be `BoundingBox`, but instead really consider x/y only. Remember that the geometry is extruded in the z-direction, so `BoundingBox` is wrong for my purpose.

I can change the name to `point_in_2d_box` if you find it more descriptive.

Reply:
But why not use the 2-d version of `BoundingBox` with `Point<2>` then? Isn't it still the same thing?

Reply:
it is the same thing and I could use it, but I would need to have two temporary objects for 2D and 3D case to do the job. I don't like this.

Reply:
Fair enough. Lets just stick with the current function, renamed to `point_in_2d_box`.